libraries/AP_Networking/AP_Networking.h

```diff
@@ -175,6 +180,51 @@ class AP_Networking
 
     HAL_Semaphore sem;
 
+    enum class NetworkPortType {
+        NONE = 0,
+        UDP_CLIENT = 1,
+    };
+
+    // class for NET_Pn_* parameters
+    class Port : public AP_SerialManager::RegisteredPort {
+    public:
+        /* Do not allow copies */
+        CLASS_NO_COPY(Port);
+
+        Port() {}
+
+        static const struct AP_Param::GroupInfo var_info[];
+        AP_Enum<NetworkPortType> type;
+        AP_Networking_IPV4 ip {"0.0.0.0"};
+        AP_Int32 port;
+        SocketAPM *sock;
+
+        bool is_initialized() override {
+            return true;
+        }
+        bool tx_pending() override {
+            return false;
+        }
+
+        void udp_client_init(void);
+        void udp_client_loop(void);
+
+    private:
+        bool init_buffers(uint32_t size);
```

Comment:
Add WARN_IF_UNUSED. If this fails, you need to call it again.

Reply:
its a private interface, we normally use WARN_IF_UNUSED on public interfaces